### ISSUE TITLE
Duplicates from Insertion Update

### ISSUE THREAD

**Issue by Stevo15025** [Stevo15025/tsp_gpu#12]
Insertion Update is closed to finished, but we are getting some duplicates which I'm not sure about. For example

![image](https://cloud.githubusercontent.com/assets/5857231/20803477/608526be-b7bd-11e6-857a-5683171efdd7.png)

I'm pretty sure it's because of 
```C
...
           if (xid >= city_one_swap && xid <= city_two_swap){
                salesman_route[xid] = salesman_route2[xid + 1];
            }           
} else {
            if (xid >= city_two_swap && xid <= city_one_swap){
                salesman_route[xid] = salesman_route2[xid - 1];
            }
...
```

https://github.com/Stevo15025/tsp_gpu/blob/master/tsp_solve.h#L345

In the insertion update step. Those may just need to be 'not equal to' on one side

**Comment by Stevo15025** [comment]
The blocks per grid for the trip should probably be

```CUDA
dim3 blocksPerTripGrid((N / t_num) + 1, 1, 1);
```

instead of 
```CUDA
dim3 blocksPerTripGrid((N+1) / t_num, 1, 1);
```


https://github.com/Stevo15025/tsp_gpu/blob/master/kernel.cu#L140

**Comment by Stevo15025** [comment]
Fixed with  5e40803

**Comment by Stevo15025** [comment]
They mainly happen at the end
```
99993      True
99994      True
99995     False
99996     False
99997      True
99998     False
99999     False
100000     True
```

Where 100,000 should be True since it is 0

**Comment by Stevo15025** [comment]
@PB12203006 

Why is there a + 1 in

```
if (xid == 0)
   salesman_route[city_two_swap + 1] = salesman_route2[city_one_swap];
```

for the second part of `tspInsertUpdate2, but not the opposite in the first?

**Comment by PB12203006** [comment]
I've updated a correct version, and this part is right. Because we insert the city_one between city_two and two+1 ,so there's a little difference between two conditions, one condition the city two is squeezed to shift, another condition the city_two remains, because city_one squeezes another direction.

**Comment by Stevo15025** [comment]
Closed with  c512258
